### DISCUSSION TITLE
Representation of OSM relations

### DISCUSSION THREAD

**Discussion by Lychfindel** [onthegomap/planetiler#61] [Q&A] [answered]
Hi!
I came across Planetiler when I was looking for vector tiles to include in my project, and I have to tell you that your script is incredibly fast and reliable! Thank you so much!

Now for the real question!
I'm not able to render some OSM relation correctly.
I'm currently using this basic style: https://raw.githubusercontent.com/De-Qua/map-style/main/style2022.json
The problem is that some places are correctly identified as polygons with class path (e.g. https://www.openstreetmap.org/relation/7932512), but other places, when exported, appear neither as polygons nor with other attributes (e.g. https://www.openstreetmap.org/relation/4224254).

Do you have any hints why there is this difference and how I might resolve it?

EDIT: I made a mistake while looking at the tiles in maputnik: using the tiles created with Planetiler all the multipolygons (or maybe the relations, I still have to understand) are missing

**Comment by msbarry** [comment, marked as answer]
Thanks for reaching out! It looks like the logic ported from openmaptiles requires pedestrian areas to have `area=yes` tag to distinguish them from a looping path, but those are multipolygons which don't need an area tag to imply it's a polygon. Fixing this in #63 

It seems to look right with that change:

![image](https://user-images.githubusercontent.com/1480504/151347694-34863b59-da6b-4bff-b171-12ff5db3cc50.png)

**Reply by Lychfindel** [reply]
Hi Michael,
Thanks a lot! I'm happy I helped to fix a bug :)
Unfortunately I'm not able to build the jar file…
I have downloaded the repo and then I just run
`mvn clean package`
But some tests are failing:
```
[ERROR] Failures:
[ERROR]   PlanetilerTests.testMetadataButNoPoints:226  others: {json={"vector_layers":[]}} ==> expected: <{attribution=test attribution, bounds=-180,-85.05113,180,85.05113, center=0,0,0, description=test description, format=pbf, maxzoom=14, minzoom=0, name=test name, type=baselayer, version=test version}> but was: <{attribution=test attribution, bounds=-180,-85,05113,180,85,05113, center=0,0,0, description=test description, format=pbf, maxzoom=14, minzoom=0, name=test name, type=baselayer, version=test version}>
[ERROR]   MbtilesTest.testAddMetadata:117 expected: <{attribution=attribution value, bounds=-90,-66.51326,90,66.51326, center=0,0,1, description=description value, format=pbf, maxzoom=13, minzoom=1, name=name value, type=baselayer, version=1.2.3}> but was: <{attribution=attribution value, bounds=-90,-66,51326,90,66,51326, center=0,0,1, description=description value, format=pbf, maxzoom=13, minzoom=1, name=name value, type=baselayer, version=1.2.3}>
[ERROR]   MbtilesTest.testAddMetadataSmallBounds:143 expected: <{bounds=-73.6632,41.1274,-69.7598,43.0185, center=-71.7115,42.07295,7}> but was: <{bounds=-73,6632,41,1274,-69,7598,43,0185, center=-71,7115,42,07295,7}>
[ERROR]   MbtilesTest.testAddMetadataWorldBounds:130 expected: <{bounds=-180,-85.05113,180,85.05113, center=0,0,0}> but was: <{bounds=-180,-85,05113,180,85,05113, center=0,0,0}>
[ERROR]   FormatTest.testFormatDecimal:69 expected: <0.1> but was: <0,1>
[ERROR]   FormatTest.testFormatDecimal:69 expected: <0.1> but was: <0,1>
[ERROR]   FormatTest.testFormatDecimal:69 expected: <1,111.1> but was: <1.111,1>
[INFO]
[ERROR] Tests run: 789, Failures: 7, Errors: 0, Skipped: 0
```

Could you please help me with the build of the jar file?

Thanks again!

**Reply by msbarry** [reply]
Hmm that's funny... what does `mvn --version` show?

**Reply by Lychfindel** [reply]
Here the output:
```
Apache Maven 3.8.4 (9b656c72d54e5bacbed989b64718c159fe39b537)
Maven home: /Library/apache-maven-3.8.4
Java version: 17.0.1, vendor: Eclipse Adoptium, runtime: /Library/Java/JavaVirtualMachines/temurin-17.jdk/Contents/Home
Default locale: it_IT, platform encoding: UTF-8
OS name: "mac os x", version: "10.13.6", arch: "x86_64", family: "mac"
```

**Reply by msbarry** [reply]
Ah, I think I've got it... the tests depend on locale-dependent number formatting. I'll fix this in #64 but in the meantime you can build with `mvn -DskipTests clean package`

**Reply by Lychfindel** [reply]
Ok without the tests the build works!
But I don't understand why, but I still don't see the polygons… I have used the file `planetiler-dist/target/planetiler-dist-0.3-SNAPSHOT-with-deps.jar`...
I'm sorry but I feel stupid since in your screenshot the polygons are correctly renderer…

![image](https://user-images.githubusercontent.com/58313635/151459471-192b33fc-63e3-40c4-a925-1c6ae7ff980e.png)

**Reply by msbarry** [reply]
You should be able to run with tests if you build from main now. Sometimes maplibre caches tiles too aggressively, you might need to clear your cache and reload to get the new tiles, or run locally on a different port. To be sure you could run with `--only-layers=transportation` to make sure the tiles your seeing are the new ones.

**Reply by Lychfindel** [reply]
Thank you very much Michael, it was indeed just a cache problem!
